common/component/kafka/metadata.go

```diff
@@ -129,25 +129,25 @@ func (k *Kafka) upgradeMetadata(metadata map[string]string) (map[string]string,
 				// If legacy authRequired was used, either SASL username or mtls is the method.
 				if saslPassPres && saslPassVal != "" {
 					// User has specified saslPassword, so intend for password auth.
-					metadata[authType] = passwordAuthType
+					meta[authType] = passwordAuthType
 				} else {
-					metadata[authType] = mtlsAuthType
+					meta[authType] = mtlsAuthType
 				}
 			} else {
-				metadata[authType] = noAuthType
+				meta[authType] = noAuthType
 			}
 		} else {
-			return metadata, errors.New("kafka error: invalid value for 'authRequired' attribute")
+			return meta, errors.New("kafka error: invalid value for 'authRequired' attribute")
 		}
 	}
 
 	// if consumeRetryEnabled is not present, use component default value
-	consumeRetryEnabledVal, consumeRetryEnabledOk := metadata[consumeRetryEnabled]
+	consumeRetryEnabledVal, consumeRetryEnabledOk := metadata.GetMetadataProperty(meta, consumeRetryEnabled)
 	if !consumeRetryEnabledOk || consumeRetryEnabledVal == "" {
-		metadata[consumeRetryEnabled] = strconv.FormatBool(k.DefaultConsumeRetryEnabled)
+		meta[consumeRetryEnabled] = strconv.FormatBool(k.DefaultConsumeRetryEnabled)
 	}
```

Comment:
The better way to do this is to set the default value in the metadata struct (not map) before calling `metadata.ParseMetadata`. It needs to be done in each component.

This way we correctly respect truthy values, and do not cause issues with casing.

Reply:
this too is preexisting. I just updated to account for metadata not clashing with the imported pkg called metadata that we're now consuming

Reply:
Could you please make this change in here since we're improving the metadata?